packages/web-app-files/src/views/spaces/GenericSpace.vue

```diff
@@ -382,6 +384,25 @@ export default defineComponent({
       }
     }
 
+    const openWithDefaultApp = () => {
+      const highlightedFile = store.getters['Files/highlightedFile']
+
+      if (!highlightedFile || highlightedFile.isFolder) {
+        return
+      }
+
+      const fileActionsOptions = {
+        resources: [highlightedFile],
+        space: props.space
+      }
+
+      const defaultEditorAction = getDefaultEditorAction(fileActionsOptions)
+
+      if (defaultEditorAction) {
+        defaultEditorAction.handler({ ...fileActionsOptions })
+      }
+    }
+
```

Comment:
@kulmann this might be also triggered when public link is opened so we check if highlightedFile is a folder and don't proceed